src/power_grid_model/core/power_grid_meta.py

```diff
@@ -17,7 +17,14 @@
 from power_grid_model.core.power_grid_core import AttributePtr, ComponentPtr, DatasetPtr, IdxPtr
 from power_grid_model.core.power_grid_core import power_grid_core as pgc
 
-_CTYPE_NUMPY_MAP = {"double": "f8", "int32_t": "i4", "int8_t": "i1", "double[3]": "(3,)f8"}
+# constant enum for ctype
+PGM_INT32 = 0
+PGM_INT8 = 1
+PGM_DOUBLE = 2
+PGM_DOUBLE3 = 3
```

Comment:
maybe it's a good idea to put this in an `IntEnum`: `class PGM_CType(IntEnum)`. You can then automatically cast from `int` to `PgmCType` and vice versa

Reply:
done